core/src/main/java/org/apache/gravitino/meta/AuditInfo.java

```diff
@@ -225,6 +226,12 @@ public Builder withLastModifiedTime(Instant lastModifiedTime) {
      * @return the constructed and validated {@link AuditInfo} instance.
      */
     public AuditInfo build() {
+      AuditInfo auditInfo = new AuditInfo();
+      auditInfo.creator = this.creator;
+      auditInfo.createTime = this.createTime;
+      auditInfo.lastModifier = this.lastModifier;
+      auditInfo.lastModifiedTime = this.lastModifiedTime;
+
```

Comment:
Another approach is to utilise the auditInfo instance and create a new one after the assignment. That would be less code/changes, but I'm not sure that's a better approach. What do you think?

Reply:
> Another approach is to utilise the auditInfo instance and create a new one after the assignment. That would be less code/changes, but I'm not sure that's a better approach. What do you think?

Thanks for the feedback! I considered that approach initially, but I think the current implementation is better for thread safet yand follows standard builder patterns. Each `build()` call would create a completely independent instance without side effects on the builder’s state. While it’s a few more lines of code, I think it would be better in the long run.

Reply:
Thanks for that.